test/extended/operators/images.go

```diff
@@ -163,16 +169,18 @@ func ignoredNamespace(namespace string) bool {
 	// TODO components in openshift-operators may not come from our payload, may want to weaken restriction
 	namespacePrefixes := sets.NewString("kube-", "openshift-")
 	ignoredNamespacePrefixes := sets.NewString("openshift-marketplace", "openshift-must-gather-")
+
+	ret := true
 	for _, prefix := range namespacePrefixes.List() {
-		if !strings.HasPrefix(namespace, prefix) {
-			return true
+		if strings.HasPrefix(namespace, prefix) {
```

Comment:
This change make the `ignoredNamespace()` function to return false for everything that start with `kube-` or `openshift-`, even for the `openshift-marketplace` and `openshift-must-gather-` prefixes we want to ignore.

The previous implementation seemed correct to me.

Reply:
@mandre Thanks for the review.  For we have multiple prefixes here, such as `kube-`, `openshift-`, then without the fix, for the example namespace `openshift-cluster-storage-operator`, it will always be ignored with the following loop(it doesn't have prefixed `kube-` and `openshift-`):
```
for _, prefix := range namespacePrefixes.List() {
		if !strings.HasPrefix(namespace, prefix) {
			return true
               }
}
```

Reply:
With this loop, my fix might have this issue too when the prefix list changed, update my patch too.

Reply:
Hmmm... I understand that the original loop ignores everything that does not start with `kube-` or `openshift-`, but still ignores `openshift-marketplace` and `openshift-must-gather-`.

As such, `openshift-cluster-storage-operator` has the `openshift-` prefix and shouldn't be ignored. Am I missing something?

Reply:
for the `openshift-cluster-storage-operator` namespace, with the profixes list ("kube-", "openshift-"), when it enters the following loop:
```
for _, prefix := range namespacePrefixes.List() {
		if !strings.HasPrefix(namespace, prefix) {
			return true
               }
}
```
It doesn't have the prefix `kube-`, then it returns true, then the ns is ignored.

Reply:
Oh, I see, you're right. The function will be clearer like the following I think:
```
// ignoredNamespace() returns true if the namespace is to be ignored by the test
func ignoredNamespace(namespace string) bool {
	// a pod in a namespace that begins with kube-* or openshift-* must come from our release payload
	// TODO components in openshift-operators may not come from our payload, may want to weaken restriction
	namespacePrefixes := sets.NewString("kube-", "openshift-")
	ignoredNamespacePrefixes := sets.NewString("openshift-marketplace", "openshift-must-gather-")

	ignore := true
	for _, prefix := range namespacePrefixes.List() {
		if strings.HasPrefix(namespace, prefix) {
			ignore = false
			break
		}
	}
	for _, prefix := range ignoredNamespacePrefixes.List() {
		if strings.HasPrefix(namespace, prefix) {
			ignore = true
			break
		}
	}
	return ignore
}
```

Reply:
@mandre Thank you! Updated.